ui/src/app/shared/edge/edgeconfig.ts

```diff
@@ -402,6 +402,21 @@ export class EdgeConfig {
         return result;
     }
 
+    /**
+     * Returns the corresponding icon for a given factory
+     */
+    public getComponentsIconByFactory(factory: EdgeConfig.Factory): string {
+        let result = null;
+        this.listAvailableFactories().forEach(availableFactories => {
+            availableFactories.factories.forEach(availableFactory => {
+                if (factory == availableFactory) {
+                    result = availableFactories.category.icon;
+                }
+            })
+        })
+        return result;
```

Comment:
Should return some empty default icon maybe?! (to be defined as default in line 409)